coil-compose-core/src/commonMain/kotlin/coil3/compose/CrossfadePainter.kt

```diff
@@ -99,12 +99,14 @@ class CrossfadePainter(
 
         val isStartSpecified = startSize.isSpecified
         val isEndSpecified = endSize.isSpecified
-        if (isStartSpecified && isEndSpecified) {
-            return Size(
-                width = maxOf(startSize.width, endSize.width),
-                height = maxOf(startSize.height, endSize.height),
-            )
+
+        if (isEndSpecified) {
+            return endSize
+        }
```

Comment:
In my opinion, we could simply set the intrinsicSize based on the endSize. What do you think?

Reply:
That would be a significant behaviour change and I'm not sure about the ramifications. How would crossfading from a painter with an intrinsic size to a painter with no intrinsic size (e.g. `ColorPainter`) inside an `AsyncImage` with no constraints look? Would it have 0 size?